PyInstaller/utils/hooks/conda.py

```diff
@@ -193,3 +194,67 @@ def collect_dynamic_libs(name, dest=".",
                 files.append((str(CONDA_ROOT / file), dest))
 
     return files
+
+
+PYTHONPATH_PREFIXES = []
+for path in sys.path:
+    path = Path(path)
+    try:
+        PYTHONPATH_PREFIXES.append(path.relative_to(sys.prefix))
+    except ValueError:
+        pass
+
+
+def _get_package_name(file):
+    """Determine the package name of a Python file in ``sys.path``.
+
+    :param file: A Python filename relative to Conda root (sys.prefix).
+    :return: Package name or None.
+
+    This function only considers single file packages e.g. ``foo.py`` or
+    top level ``foo/__init__.py``\\ s. Anything else is ignored (returning
+    ``None``).
+    """
+    file = Path(file)
+    if file.name == "__init__.py":
+        file = file.parent
+
+    elif not (file.suffix in compat.ALL_SUFFIXES or file.is_dir()):
+        return
+
+    for prefix in PYTHONPATH_PREFIXES:
+        if len(file.parts) != len(prefix.parts) + 1:
+            continue
+        if fnmatch.fnmatch(str(file.parent), str(prefix)):
```

Comment:
Why using fnmatch here? Please add a short comment explaining.
Doesn't the following work?
```
if file.parent.parts == prefix.parts:
```